controller/cache/cache.go

```diff
@@ -348,20 +352,37 @@ func skipResourceUpdate(oldInfo, newInfo *ResourceInfo) bool {
 		return false
 	}
 	isSameHealthStatus := (oldInfo.Health == nil && newInfo.Health == nil) || oldInfo.Health != nil && newInfo.Health != nil && oldInfo.Health.Status == newInfo.Health.Status
-	isSameManifest := oldInfo.manifestHash != "" && newInfo.manifestHash != "" && oldInfo.manifestHash == newInfo.manifestHash
+	isSameManifest := oldInfo.manifestHash == newInfo.manifestHash
```

Comment:
This condition should not change. If a resource does not have a hash, that means the annotation was not there (or not true), therefore we should not ignore updates.

Reply:
Or, you want the dependent object set `argocd.argoproj.io/apply-resources-update=true` and then use the existing ignore resources config to skip refresh?